Themis/Views/Assessment/CorrectionSidebar/CorrectionSidebarView.swift

```diff
@@ -30,24 +30,14 @@ struct CorrectionSidebarView: View {
         assessmentVM.participation?.getExercise(as: ProgrammingExercise.self)?.templateParticipation?.id
     }
     
-    init(assessmentResult: Binding<AssessmentResult>,
-         assessmentVM: AssessmentViewModel,
-         cvm: CodeEditorViewModel,
-         courseId: Int) {
-        self._assessmentResult = assessmentResult
-        self.assessmentVM = assessmentVM
-        self.cvm = cvm
-        self.courseId = courseId
-    }
-    
     var body: some View {
         VStack {
             sideBarElementPicker
             
             if !assessmentVM.loading {
                 ZStack {
                     ScrollView {
-                        ProblemStatementView(courseId: courseId, exerciseId: exercise?.id ?? -1)
+                        ProblemStatementView(courseId: courseVM.shownCourseID ?? -1, exerciseId: exercise?.id ?? -1)
```

Comment:
Suggestion: Maybe have an `if let` above and then show some kind of short error text if the `shownCourseID` or the `exercise.id` is `nil` respectively